dot/network/notifications.go

```diff
@@ -315,6 +303,11 @@ func (s *Service) sendData(peer peer.ID, hs Handshake, info *notificationsProtoc
 			closeOutboundStream(info, peer, stream)
 		}
 		return
+	} else if s.host.messageCache != nil {
+		if _, err := s.host.messageCache.put(peer, msg); err != nil {
+			logger.Errorf("failed to add message to cache for peer %s: %s", peer, err)
+			return
```

Comment:
Do we want to return here if we fail to put it in the cache? Or should we let it continue? :thinking:

Reply:
Unfortunately, I don't have a clear answer to this.

The difference between returning and not returning is execution of ReportPeer. 
- I you don't return ReportPeer will get executed multiple times (increasing the reputation every time) until we are able to put the message in cache. 

- If we do return, ReportPeer does not get executed, until we are able to put the message in cache, increasing the reputation only once.

None of these options are good, but the second one seems less bad. Also, we were returning previously as well, so this behavior has not changed.

However, if we fail to add msg to cache, in both cases, we have a bigger problem, that we keeping sending the same message, and our reputation decreases.